staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -622,3 +531,163 @@ func TestInvalidKeys(t *testing.T) {
+					},
+				},
+			},
+			Status: corev1.NodeStatus{
+				Allocatable: corev1.ResourceList{
+					corev1.ResourceCPU:     *resource.NewMilliQuantity(10, resource.DecimalSI),
+					corev1.ResourceMemory:  *resource.NewQuantity(20*(1024*1024), resource.DecimalSI),
+					corev1.ResourceStorage: *resource.NewQuantity(30*(1024*1024), resource.DecimalSI),
+				},
+			},
+		}
+	}
+	testCases := []struct {
+		name              string
+		objectNum         int
+		newObjectFunc     func() runtime.Object
+		newListObjectFunc func() runtime.Object
+	}{
+		{
+			name:              "list 10 corev1.Pod",
+			objectNum:         10,
+			newObjectFunc:     generatePod,
+			newListObjectFunc: func() runtime.Object { return &corev1.PodList{} },
+		},
+		{
+			name:              "list 100 corev1.Pod",
+			objectNum:         100,
+			newObjectFunc:     generatePod,
+			newListObjectFunc: func() runtime.Object { return &corev1.PodList{} },
+		},
+		{
+			name:              "list 1000 corev1.Pod",
+			objectNum:         1000,
+			newObjectFunc:     generatePod,
+			newListObjectFunc: func() runtime.Object { return &corev1.PodList{} },
+		},
+		{
+			name:              "list 10000 corev1.Pod",
+			objectNum:         10000,
+			newObjectFunc:     generatePod,
+			newListObjectFunc: func() runtime.Object { return &corev1.PodList{} },
+		},
+		{
+			name:              "list 50000 corev1.Pod",
+			objectNum:         50000,
+			newObjectFunc:     generatePod,
+			newListObjectFunc: func() runtime.Object { return &corev1.PodList{} },
+		},
+		{
+			name:              "list 10 corev1.Node",
+			objectNum:         10,
+			newObjectFunc:     generateNode,
+			newListObjectFunc: func() runtime.Object { return &corev1.NodeList{} },
+		},
+		{
+			name:              "list 100 corev1.Node",
+			objectNum:         100,
+			newObjectFunc:     generateNode,
+			newListObjectFunc: func() runtime.Object { return &corev1.NodeList{} },
+		},
+		{
+			name:              "list 1000 corev1.Node",
+			objectNum:         1000,
+			newObjectFunc:     generateNode,
+			newListObjectFunc: func() runtime.Object { return &corev1.NodeList{} },
+		},
+		{
+			name:              "list 10000 corev1.Node",
+			objectNum:         10000,
+			newObjectFunc:     generateNode,
+			newListObjectFunc: func() runtime.Object { return &corev1.NodeList{} },
+		},
+		{
+			name:              "list 50000 corev1.Node",
+			objectNum:         50000,
+			newObjectFunc:     generateNode,
+			newListObjectFunc: func() runtime.Object { return &corev1.NodeList{} },
+		},
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			// booting etcd instance
+			ctx, store, etcdClient := testSetup(b)
+			defer etcdClient.Close()
+
+			// make fake objects..
+			dir := "/testing"
+			key := dir + "/testkey"
+			originalRevision := ""
+			for i := 0; i < tc.objectNum; i++ {
+				obj := tc.newObjectFunc()
+				o := obj.(metav1.Object)
+				o.SetName(fmt.Sprintf("object-%d", i))
+				o.SetSelfLink("testlink")
```

Comment:
Let's not set SelfLink - selflinks are deprecated

Reply:
fixed